electrum/plugins/trezor/qt.py

```diff
@@ -767,3 +793,110 @@ def slider_released():
+        client = self.plugins.device_manager.client_by_id(device_id, scan_now=False)
+        if not client.is_uptodate():
+            msg = (_('Outdated {} firmware for device labelled {}. Please '
+                     'download the updated firmware from {}')
+                     .format(_info.model_name, _info.label, self.plugin.firmware_URL))
+            self.error = msg
+            return
+
+        message = _('Choose how you want to initialize your {}.').format(_info.model_name)
+        choices = [
+            # Must be short as QT doesn't word-wrap radio button text
+            (TIM_NEW, _("Let the device generate a completely new seed randomly")),
+            (TIM_RECOVER, _("Recover from a seed you have previously written down")),
+        ]
+        self.choice_w = ChoiceWidget(message=message, choices=choices)
+        self.layout().addWidget(self.choice_w)
+        self.layout().addStretch(1)
+
+        self._valid = True
+
+    def apply(self):
+        self.wizard_data['trezor_init'] = self.choice_w.selected_item[0]
+
+
+class WCTrezorInitParams(WizardComponent):
+    def __init__(self, parent, wizard):
+        WizardComponent.__init__(self, parent, wizard, title=_('Set-up trezor'))
+        self.plugins = wizard.plugins
+        self._busy = True
+
+    def on_ready(self):
+        _name, _info = self.wizard_data['hardware_device']
+        self.settings_layout = InitSettingsLayout(self.plugins.device_manager, self.wizard_data['trezor_init'], _info.device.id_)
+        self.layout().addLayout(self.settings_layout)
+        self.layout().addStretch(1)
+
+        self.valid = True
+        self.busy = False
+
+    def apply(self):
+        self.wizard_data['trezor_settings'] = self.settings_layout.get_settings()
+
+
+class WCTrezorInit(WizardComponent, Logger):
+    def __init__(self, parent, wizard):
+        WizardComponent.__init__(self, parent, wizard, title=_('Set-up trezor'))
+        Logger.__init__(self)
+        self.plugins = wizard.plugins
+        self.plugin = self.plugins.get_plugin('trezor')
+
+        self.layout().addWidget(WWLabel('Done'))
+
+        self._busy = True
+
+    def on_ready(self):
+        settings = self.wizard_data['trezor_settings']
+        method = self.wizard_data['trezor_init']
+        _name, _info = self.wizard_data['hardware_device']
+        device_id = _info.device.id_
+        client = self.plugins.device_manager.client_by_id(device_id, scan_now=False)
+        client.handler = self.plugin.create_handler(self.wizard)
+
+        def initialize_device_task(settings, method, device_id, handler):
+            self.plugin._initialize_device(settings, method, device_id, handler)
+            self.init_done()
+
+        t = threading.Thread(
+            target=initialize_device_task,
+            args=(settings, method, device_id, client.handler),
+            daemon=True)
+        t.start()
+
+    def init_done(self):
+        self.logger.info('Done initialize device')
```

Comment:
I am testing with a trezor one, uninitialized.
Regardless of generating a new random seed, or restoring from existing seed, the wizard gets stuck after last word, just showing "done". The next button is disabled.
![pic2](https://github.com/spesmilo/electrum/assets/29142493/d76a5423-f17e-42a1-b818-e1d0baa577f2)

note: does not affect keepkey

Reply:
I hit exactly the same issue with keepkey, see this comment: https://github.com/spesmilo/electrum/pull/8560#issuecomment-1717968858
Apparently it was fixed for keepkey, but not for other devices?

Reply:
Should be fixed in b6fdd128a1e49bff1dfa10223be3f46e6747a354